pkg/controller/operators/operator.go

```diff
@@ -0,0 +1,38 @@
+package operators
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorlister"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/queueinformer"
+)
+
+// Operator contains shared operator tooling.
+type Operator struct {
+	*queueinformer.Operator
+
+	Client       versioned.Interface
+	Lister       operatorlister.OperatorLister
+	clock        clock.Clock
+	namespaces   []string
+	resyncPeriod time.Duration
+}
+
+// Now returns the operator's view of the current time.
+func (o *Operator) Now() metav1.Time {
+	return metav1.NewTime(o.clock.Now())
+}
+
+// Namespaces returns the list namespaces the operator is configured to watch.
+func (o *Operator) Namespaces() []string {
```

Comment:
is this conforming to some interface we can mock? why have the accessor?

Reply:
This is meant to be an embeddable base for more complex operators. The accessors expose useful private fields to those operators. This was supposed to protect from an invalid initialization - though looking at it now, I think pushing them into an interface and having an entirely private struct that implements them is probably a better path. How does that sound instead?

Reply:
Either that or having them just be public members seems fine - depends on where you see the re-use being.